src/udev/udevadm-wait.c

```diff
@@ -0,0 +1,363 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
+
+#include <getopt.h>
+#include <unistd.h>
+
+#include "sd-event.h"
+
+#include "alloc-util.h"
+#include "chase-symlinks.h"
+#include "device-util.h"
+#include "errno-util.h"
+#include "fd-util.h"
+#include "inotify-util.h"
+#include "parse-util.h"
+#include "path-util.h"
+#include "static-destruct.h"
+#include "strv.h"
+#include "udev-util.h"
+#include "udevadm.h"
+
+static usec_t arg_timeout_usec = USEC_INFINITY;
+static bool arg_initialized = true;
+static bool arg_removed = false;
```

Comment:
these two should be a single enum, no? i.e.

```c
enum {
        ARG_WAIT_UNTIL_ADDED,
        ARG_WAIT_UNTIL_INITIALIZED,
        ARG_WAIT_UNTIL_REMOVED,
} arg_wait_until = ARG_INITIALIZED;
```

no? and then maybe also expose it like this in the cmdline, i.e. `udevadm wait --until=added /dev/sda1` or so?